rfcs/0010-custom-system-uris.md

```diff
@@ -0,0 +1,325 @@
+
+_Changing_ the URIs in v1 or v2 would be a major backwards-incompatibility,
+  it would break almost all existing customer applications,
+  which is something we strive very hard to avoid doing.
+
+Additionally, it's hard to come up with a different answer that is both future-proof **and** would deliver any real user value,
+  even in a future version of the API, e.g. v3.
+* A base URI such as `https://bfd.dasg.oeda.cms.gov/` would capture details of CMS' _current_ internal organizational chart,
+    in ways that would be awkward in the face of potential future re-organizations.
+  If, for example, the Data and Analytics Strategy Group (DASG) gets renamed,
+    would we want to change the URI again?
+* A base URI such as `https://claims-data-api.bfd.cms.gov/` is perhaps a safer and more ontologically-correct choice,
+    but it's hard to imagine any existing user caring such that they'd feel it justifies changing
+    the URIs already baked into the v1/v2 version of their application.
+  It's also perhaps a bit of an organizational overreach,
+    as BFD is hardly the only "claims data API" at CMS.
+* A base URI such as `https://bfd.cms.gov/` is perhaps a safer and more ontologically-correct choice,
+    but it's hard to imagine any existing user caring such that they'd feel it justifies changing
+    the URIs already baked into the v1/v2 version of their application.
+
+
+### Proposed Solution: Detailed Design
+[Proposed Solution: Detailed Design]: #proposed-solution-detailed-design
+
+<!--
+This is the technical portion of the RFC. Explain the design in sufficient detail that:
+
+* Its interaction with other features is clear.
+* It is reasonably clear how the feature would be implemented.
+* Corner cases are dissected by example.
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
+-->
+
+Taking a look at BFD's
+  [TransformerConstants](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/TransformerConstants.java)
+  file reveals the following base URI structure:
+
+```
+https://bluebutton.cms.gov/resources/
+  codesystem/
+    adjudication/
+      <various EOB.adjudication.category discriminators>
+    information/
+      <various EOB.information.category discriminators>
+    variables/
+      <all sorts of stuff, but mostly EOB.extension discriminators>
+  identifier/
+    <several identifier.system discriminators>
+```
+
+_Most_ of the leaf URI "path" components refer to variables found in the
+  [CCW Data Dictionaries](https://www2.ccwdata.org/web/guest/data-dictionaries).
+As such, the URI structure should be extended for incorporation of future, non-CCW fields.
+For PACA, something like the following would be recommended,
+  which would ensure proper namespacing of similar-yet-actually-distinct fields:
```

Comment:
@karlmdavis when we have a strong inclination that like data elements in CCW are the same as those in a shared system (without production data some of these are hard to confirm, but when we compare data dictionary values between CCW and the shared systems, the descriptions are identical).  Should we reuse the existing CCW based system or create a new one for the shared system?

For example, our FISS `LOB-CD` enum description and possible values matches that of https://bluebutton.cms.gov/resources/variables/clm_fac_type_cd/.  Which of the following are you recommending?
- https://bluebutton.cms.gov/resources/variables/clm_fac_type_cd/
- https://bluebutton.cms.gov/resources/variables/fiss/lob-cd/

Reply:
I understand the reason for option number 2, it feels safer having no production partially-adjudicated data to confirm these assumptions with.  However, as to future consolidation with additional codings.  Will that only work with CodeableConcepts?  What about extensions, would we just duplicate the value twice in the extensions list so it would show up as two different systems?

Reply:
Yeah, I think that'd be how you'd have to do it -- short of a breaking change, leastways.

Reply:
which is the strategy we've been implementing when we hit something like that; add in the Codable where it is expected, but also add it as an extension. Sure it's a duplication of data, but it allows backwards compatibility.